src/Instance/InstanceDetails/InstanceDescriptiveView/utils.js

```diff
@@ -1,8 +1,19 @@
-import languagetable from '../../../data/languages';
+import { useIntl } from 'react-intl';
+
+import {
+  intlPreferredLanguageCode,
+  languageByCode,
+} from '@folio/stripes/components';
 
 // eslint-disable-next-line
 export const formatLanguages = (languages = []) => {
+  const intl = useIntl();
   return languages
-    .map(languageCode => languagetable.languageByCode(languageCode))
+    .map(languageCode => {
+      const intlCode = intlPreferredLanguageCode(languageCode);
+      // If formatDisplayName returns a localized language name, great -- use that.
+      // Otherwise, fall back to the English name.
+      return intl.formatDisplayName(intlCode, { fallback: 'none' }) || languageByCode(languageCode).name;
```

Comment:
This same logic shows up here and in `InstanceFilters`. Is it worth wrapping into a function and exporting along with the other new functions you added in stripes-components?

Reply:
I was not aware of `intl.formatDisplayName`. Nice work @Baroquem. I Agree with @zburke maybe this belongs to `stripes-components`?

Reply:
@zburke @mkuklis You're both right, of course. I've done some cleanup and tried to simplify the whole setup. Michal, I wasn't aware of formatDisplayName either; 'twas Zak who pointed me in that direction!